index.js

```diff
@@ -17,10 +27,67 @@ if (!fs.existsSync(backlightPath)) {
     throw new Error('Backlight control not supported (' + backlightPath + ' does not exist)');
 }
 
-exports.powerOn = function () {
+/** Power managment */
+exports.powerOn = () => {
     return writeValue('bl_power', '0');
 };
 
-exports.powerOff = function () {
+exports.powerOff = () => {
     return writeValue('bl_power', '1');
 };
+
+exports.isPoweredOn = () => {
+    return new Promise((resolve, reject) => {
+        readValue('bl_power').then((powerValue) => {
+            resolve(parseInt(powerValue) === 0 ? true : false);
+        }, (err) => {
+            reject(err);
+        }); 
+    });
+};
+
+/** Brightness managment */
+exports.getBrightness = () => {
+    return new Promise((resolve, reject) => {
+        readValue('actual_brightness').then((brightnessValue) => {
+            resolve(parseInt(brightnessValue));
+        }, (err) => {
+            reject(err);
+        }); 
+    });
+};
+
+exports.setBrightness = (value) => {
+    return new Promise((resolve, reject) => {
+        this.getMaxBrightness().then((maxBrightnessValue) => {
+            if (value > maxBrightnessValue || value < 0) {
+                resolve(false);
+            } else {
+                resolve(writeValue('brightness', value));
+            }            
+        }, (err) => {
+            reject(err);
+        });
+    });
+};
+
+exports.getMaxBrightness = () => {
+    return new Promise((resolve, reject) => {
+        readValue('max_brightness').then((maxBrightnessValue) => {
+            resolve(parseInt(maxBrightnessValue));
+        }, (err) => {
+            reject(err);
+        });
+    });
+};
+
+exports.setMaxBrightness = () => {
+    return new Promise((resolve, reject) => {
+        this.getMaxBrightness().then((maxBrightnessValue) => {
+            writeValue('brightness', maxBrightnessValue);
```

Comment:
This should be `max_brightness`. Actually though, are you even allowed to set max_brightness? Is that a thing? What does it do?

Reply:
It would seem that the name of the function does not correspond to what it really does. It does not set the max_brightness but sets the brightness to the value written in the max_brightness file. Do you have an idea to have a more correct function name?

For the max_brightness file it is not allowed to modify it.

Reply:
Ahh, I see. I think there are two good options:

* Just drop this method - it's not really necessary since you can do the same thing without it by hand.
* Change to a more general API, which might make it a bit clearer. `setBrightnessPercentage(n)`, that takes a number from 0 to 100, and sets the brightness as a proportion of the max?

Reply:
For the moment I renamed the function to: "setToMaxBrightness"

Reply:
Yep, good idea!

For now, I will remove the function and I could implement this in a second PR, may be in connection with the gradual increase / decrease of brightness?

Reply:
Sure, that's the simplest option, let's do that.